src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -764,15 +763,169 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
+            emit->emitIns_R(INS_setne, EA_1BYTE, targetReg);
+            emit->emitIns_R_R(INS_movzx, EA_1BYTE, targetReg, targetReg);
+            break;
+        }
+
+        case NI_SSE2_ConvertScalarToVector128Double:
+        {
+            assert(baseType == TYP_INT || baseType == TYP_LONG || baseType == TYP_FLOAT);
+            assert(op1 != nullptr);
+            assert(op2 != nullptr);
+            op2Reg          = op2->gtRegNum;
+            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+            emit->emitIns_SIMD_R_R_R(ins, emitTypeSize(TYP_SIMD16), targetReg, op1Reg, op2Reg);
+            break;
+        }
+
+        case NI_SSE2_ConvertScalarToVector128Int64:
+        case NI_SSE2_ConvertScalarToVector128UInt64:
+        {
+            assert(baseType == TYP_LONG || baseType == TYP_ULONG);
+            assert(op1 != nullptr);
+            assert(op2 == nullptr);
+            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+            // TODO-XArch-CQ -> use of type size of TYP_SIMD16 leads to
+            // instruction register encoding errors for SSE legacy encoding
+            emit->emitIns_R_R(ins, emitTypeSize(baseType), targetReg, op1Reg);
+            break;
+        }
+
+        case NI_SSE2_ConvertScalarToVector128Single:
+        {
+            assert(baseType == TYP_DOUBLE);
+            assert(op1 != nullptr);
+            assert(op2 != nullptr);
+            op2Reg          = op2->gtRegNum;
+            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+            emit->emitIns_SIMD_R_R_R(ins, emitTypeSize(TYP_SIMD16), targetReg, op1Reg, op2Reg);
+            break;
+        }
+
+        case NI_SSE2_ConvertToDouble:
+        {
+            assert(op2 == nullptr);
+            if (op1Reg != targetReg)
+            {
+                instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+                emit->emitIns_R_R(ins, emitTypeSize(targetType), targetReg, op1Reg);
+            }
+            break;
+        }
+
+        case NI_SSE2_ConvertToInt32:
+        case NI_SSE2_ConvertToInt64:
+        case NI_SSE2_ConvertToUInt32:
+        case NI_SSE2_ConvertToUInt64:
+        {
+            assert(op2 == nullptr);
+            assert(baseType == TYP_DOUBLE || baseType == TYP_FLOAT || baseType == TYP_INT || baseType == TYP_UINT ||
+                   baseType == TYP_LONG || baseType == TYP_ULONG);
+            if (op1Reg != targetReg)
+            {
+                instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+                if (baseType == TYP_DOUBLE || baseType == TYP_FLOAT)
+                {
+                    emit->emitIns_R_R(ins, emitTypeSize(targetType), targetReg, op1Reg);
+                }
+                else
+                {
+                    emit->emitIns_R_R(INS_mov_xmm2i, emitActualTypeSize(baseType), op1Reg, targetReg);
```

Comment:
Why isn't this one using `insOfHWIntrinsic`?

Reply:
It's due to #16322 which I plan to fix in #16329

Reply:
Not sure I follow. We are using `INS_mov_xmm2i` in both the `TYP_LONG` and `TYP_ULONG` case, so it shouldn't matter which base type is used in the lookup...

That is, if only `TYP_LONG` is being set, we can fill that column of the `ConvertToUInt64` table entry with a `TODO-XArch-Bug` note that it is a workaround.

Reply:
> if only `TYP_LONG` is being set, we can fill that column of the `ConvertToUInt64` table entry with a `TODO-XArch-Bug` note that it is a workaround

It should be done for `TYP_UINT` as well. In principle proposed change is equivalent workaround to that used in current code as we always select `INS_mov_xmm2i`. The workaround currently requires one change but in case we would do it in table we should change instruction selection for multiple intrinsics having unsigned integral base types. 

If you agree that current solution is simpler I will add `TODO-XArch-Bug` comment here.

Reply:
Looks like this problem is from `baseType = JITtype2varType(sig->retType);`. We should have a new signe-awared function (e.g., `JITType2BasevarType`, or other name) that only used for HW intrinsics.

Reply:
> We should have a new sign-awared function (e.g., JITtype2BaseVarType, or other name) that only used for HW intrinsics.

Yes, it is tracked by #16329

Reply:
@tannergooding Could you reply to my comment https://github.com/dotnet/coreclr/pull/16237#discussion_r167451230